core/src/main/java/org/apache/calcite/sql/type/ReturnTypes.java

```diff
@@ -773,6 +791,28 @@ public static SqlCall stripSeparator(SqlCall call) {
     return null;
   };
 
+  public static final SqlReturnTypeInference DECIMAL_OR_DOUBLE = opBinding -> {
```

Comment:
why have both DOUBLE_OR_DECIMAL and DECIMAL_OR_DOUBLE?

Reply:
Removed `DOUBLE_OR_DECIMAL`.